cpp/src/traversal/bfs.cu

```diff
@@ -471,14 +471,16 @@ namespace cugraph {
 } // end namespace cugraph
 
 gdf_error gdf_bfs(gdf_graph *graph, gdf_column *distances, gdf_column *predecessors, int start_vertex, bool directed) {
-  GDF_REQUIRE(graph->adjList != nullptr || graph->edgeList != nullptr, GDF_INVALID_API_CALL);
-  gdf_error err = gdf_add_adj_list(graph);
+  //GDF_REQUIRE(graph->adjList != nullptr || graph->edgeList != nullptr, GDF_INVALID_API_CALL);
+
+  /*gdf_error err = gdf_add_adj_list(graph);
```

Comment:
-> Remove adding adj list, already added in wrapper before calling gdf_bfs.
-> Remove int32 checks as it is ensured when adding graph itself.
-> Remove distance and predecessor column checks since the column is initialized in the bfs_wrapper.pyx with pre-set specifications(size, dtype as int32, data initialized to zeros, etc) which is redundantly verified in cpp.

Reply:
We should keep distance and predecessor column checks in case someone calls it from the c++ API. It will also help debugging if we introduce an issue at the python level.
```
  GDF_REQUIRE(distances->dtype == GDF_INT32, GDF_UNSUPPORTED_DTYPE);
  GDF_REQUIRE(predecessors->dtype == GDF_INT32, GDF_UNSUPPORTED_DTYPE);	  GDF_REQUIRE(predecessors->dtype == GDF_INT32, GDF_UNSUPPORTED_DTYPE);```